torrent/pieces.py

```diff
@@ -0,0 +1,107 @@
+from enum import Enum
+from hashlib import sha1
+import logging
+import asyncio
+
+logger = logging.getLogger(__name__)
+
+class PieceStatus(Enum):
+    free = 0 
+    done = 2
+    in_progress = 1
+
+class Piece():
+    """The piece object reprsent one piece from pieces in the info of torrent file
+    all piece in size of piece length fro torrent file hoz me except from the last piece
+    the piece come with info hash of 20 bytes in sha1 and if the piece data the came
+    from peer in sha1 not the same like the piece hash the piece is not right"""
+    def __init__(self, piece_hash, length, piece_index):
```

Comment:
i think you should create a function `set_status` that makes sure the status is from PieceStatus enum